validator/client/runner.go

```diff
@@ -305,6 +306,33 @@ func runHealthCheckRoutine(ctx context.Context, v iface.Validator, eventsChan ch
 				return
 			}
 			isHealthy := tracker.CheckHealth(ctx)
+			if !isHealthy && len(hosts) > 1 && features.Get().EnableBeaconRESTApi {
+				for i, url := range hosts {
+					if url == v.RetrieveHost() {
+						next := (i + 1) % len(hosts)
+						log.Infof("Beacon node at %s is not responding, switching to %s", url, hosts[next])
+						v.UpdateHost(hosts[next])
+
+						km, err := v.Keymanager()
+						if err != nil {
+							log.WithError(err).Fatal("Could not get keymanager")
```

Comment:
something doesn't feel right here about logging a fatal, it's needed for the push proposer settings but not sure if the whole thing should break if it can't get the keymanager

Reply:
I would say not being able to obtain the keymanager is very bad, but yeah maybe just log an error and return? Not 100% sure. This should realistically never happen anyway.